### DISCUSSION TITLE
Deprecate one of the methods of passing yip configs to kairos

### DISCUSSION THREAD

**Discussion by Itxaka** [kairos-io/kairos#1321] [Ideas]
Currently there is 3 ways that kairos runs yip config files:

 - Files in the dirs [/system/oem /oem/ /usr/local/cloud-config/]
 - cmdline with the stanza `cos.setup=`
 - cmdline, no stanza

IMO, this is too much, we should deprecate and lose the latest one. All stuff in the cmdline should be passed via the `cos.setup=` method, like everything else in the world that its passed on the cmdline.

In fact, while there is support for the last item in elemental-cli, there is no docs for it nor is probably anyone using it. Should be pretty easy to just drop it off and keep the `cos.setup=` for overrides.

Thougths?

**Comment by mauromorales** [comment]
Deprecation of the third option seems all right to me. As a newcomer to the project, I'd prefer the more specific any configuration has to be, i.e., the less magic happens, the better. And for any case where magic is decided on because it gives some nice UX, it would be nice to have detailed logs that tell the user about the whole thing.
